src-java/connecteddevices-topology/connecteddevices-storm-topology/src/main/java/org/openkilda/wfm/topology/connecteddevices/service/PacketService.java

```diff
@@ -75,52 +80,70 @@ public PacketService(PersistenceManager persistenceManager) {
      * Handle LLDP info data.
      */
     public void handleLldpData(LldpInfoData data) {
-        transactionManager.doInTransaction(() -> {
-
-            FlowRelatedData flowRelatedData = findFlowRelatedData(data);
-            if (flowRelatedData == null) {
-                return;
-            }
+        FlowRelatedData flowRelatedData = findFlowRelatedData(data);
+        if (flowRelatedData == null) {
+            return;
+        }
 
-            SwitchConnectedDevice device = getOrCreateLldpDevice(data, flowRelatedData.originalVlan);
+        String uniqueIndex = createUniqueLldpIndex(data, flowRelatedData.getOriginalVlan());
+        SwitchConnectedDevice cachedDevice = switchConnectedDeviceCache.get(uniqueIndex);
+        SwitchConnectedDevice device;
 
-            if (device == null) {
-                return;
-            }
+        if (cachedDevice != null) {
+            device = cachedDevice;
+        } else {
+            Optional<SwitchConnectedDevice> optionalDevice = switchConnectedDeviceRepository
+                    .findLldpByUniqueIndex(uniqueIndex);
+            device = optionalDevice.orElse(null);
+        }
```

Comment:
too many nulls. I'd make these lines 89-98 shorter, something like: 
```
SwitchConnectedDevice device = Optional.ofNullable(switchConnectedDeviceCache.get(uniqueIndex))
                .orElseGet(() -> switchConnectedDeviceRepository.findLldpByUniqueIndex(uniqueIndex).orElse(null));
```
the same for similar usage. (Could be even combined with the next if-else, but I don't know how to do that nicely using java 8)

Reply:
It is a great idea. Will update. Thank you!